src/main/java/winstone/accesslog/SimpleAccessLogger.java

```diff
@@ -95,44 +95,40 @@ public SimpleAccessLogger(String webAppName, Map<String, String> startupArgs) th
 
     @Override
     public void log(Request request, Response response) {
-        String uriLine = request.getMethod() + " " + request.getRequestURI() + " " + request.getProtocol();
+        String uriLine = request.getMethod() + " " + request.getHttpURI().getPath() + " "
+                + request.getConnectionMetaData().getProtocol();
         int status = response.getStatus();
-        long size = response.getContentCount();
+        long size = Response.getContentBytesWritten(response);
         String date;
         synchronized (DF) {
             date = DF.format(new Date());
         }
-
+        // https://github.com/jetty/jetty.project/blob/c5b2533fdecce21b54c6fbaf36f79bc3ba909775/jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/CustomRequestLog.java#L1093-L1099
+        Request.AuthenticationState authenticationState = Request.getAuthenticationState(request);
+        Principal principal = authenticationState == null ? null : authenticationState.getUserPrincipal();
+        String remoteUser = principal == null ? "-" : principal.getName();
         // mimic
-        // https://github.com/eclipse/jetty.project/blob/jetty-9.4.0.v20161208/jetty-server/src/main/java/org/eclipse/jetty/server/AbstractNCSARequestLog.java#L130
-        Authentication authentication = request.getAuthentication();
-        String remoteUser;
-        if (authentication instanceof Authentication.User) {
-            Authentication.User user = (Authentication.User) authentication;
-            remoteUser = user.getUserIdentity().getUserPrincipal().getName();
-        } else {
-            remoteUser = null;
-        }
-
+        // https://github.com/jetty/jetty.project/blob/3632a57f2796a2ab4fcdbfe62837d494bcd4e94a/jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/CustomRequestLog.java#L307
```

Comment:
Is the reference to the NCSA format specific to the Jetty 12 upgrade, or a general improvement? If a general improvement, can it be pushed to the default branch in anticipation of this PR, thereby shrinking the size of the diff?